yt_dlp/extractor/soundcloud.py

```diff
@@ -769,12 +778,17 @@ class SoundcloudUserIE(SoundcloudPagedPlaylistBaseIE):
     def _real_extract(self, url):
         mobj = self._match_valid_url(url)
         uploader = mobj.group('user')
+        user_id = mobj.group('user_id')
 
-        user = self._download_json(
-            self._resolv_url(self._BASE_URL + uploader),
-            uploader, 'Downloading user info', headers=self._HEADERS)
+        if user_id:
+            user_info_url = self._resolv_url(self._API_V1_BASE + 'users/' + user_id)
+            resource = 'all'
+        else:
+            user_info_url = self._resolv_url(self._BASE_URL + uploader)
+            resource = mobj.group('rsrc') or 'all'
```

Comment:
Since it's a completely different URL pattern and code path, better to create a new class